charts/redpanda/templates/statefulset.yaml

```diff
@@ -215,83 +215,38 @@ spec:
             preStop:
               exec:
                 command: ["/bin/bash", "-c", "./var/lifecycle/preStop.sh"]
+          # the startupProbe checks to see that the admin api is listening and that the broker has a node_id assigned. This check is only used to delay the start of the liveness and readiness probes until it passes.
           startupProbe:
             exec:
               command:
                 - /bin/sh
                 - -c
-                {{- if (include "admin-internal-tls-enabled" . |fromJson).bool }}
-                - >
-                  curl https://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview
-                  -svk --cacert /etc/tls/certs/{{ .Values.listeners.admin.tls.cert }}/ca.crt |
-                  awk '{
-                    id = $0; gsub(/.*"controller_id": /, "", id); gsub(/,.*/, "", id)
-                    nd_str = $0; gsub(/.*"nodes_down": \[/, "", nd_str); gsub(/\].*/, "", nd_str)
-                    FS=","
-                    split(nd_str, nd_list)
-                    for (i in nd_list) nodes_down[nd_list[i]]=""
-                    exit (id in nodes_down)
-                  }'
-                {{- else }}
-                - >
-                  curl -sv http://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview |
-                  awk '{
-                    id = $0; gsub(/.*"controller_id": /, "", id); gsub(/,.*/, "", id)
-                    nd_str = $0; gsub(/.*"nodes_down": \[/, "", nd_str); gsub(/\].*/, "", nd_str)
-                    FS=","
-                    split(nd_str, nd_list)
-                    for (i in nd_list) nodes_down[nd_list[i]]=""
-                    exit (id in nodes_down)
-                  }'
-                {{- end }}
+                - curl --silent --fail -k {{ include "admin-tls-curl-flags" . }} "{{ include "admin-http-protocol" . }}://localhost:{{ .Values.listeners.admin.port }}/v1/node_config" | grep -o '"node_id":[0-9]\+'
```

Comment:
Perhaps this check should be followed by a call to `/v1/ready` to check the status is `ready`.

As-written, this startupProbe will complete slightly before the node is actually ready to serve Kafka requests.

The startup order is:
- Bootstrap (including getting a node ID if necessary)
- Start most other services (including the admin server, which will return "booting" from /v1/ready at this point)
- Start listening on the kafka port
- Finally, set the status to "ready" in the result of /v1/ready.
- Bootstrap (this includes getting a node ID if it doesn't already have one)

Reply:
@jcsp would it be enough just to wait for the /v1/ready to both respond and give an http 200 response?

Reply:
Looks like a "no". It gives 200 for "booting" which is not "ready".

Reply:
I just switched to the ready link. That seems to be sufficient for the startupProbe

Reply:
had to use `/v1/status/ready` as `/v1/ready` was giving 404s.